engine/src/main/java/org/camunda/bpm/engine/impl/cmd/SaveAuthorizationCmd.java

```diff
@@ -65,4 +75,36 @@ public Authorization execute(CommandContext commandContext) {
     return authorization;
   }
 
+  protected void provideRemovalTime(Supplier<HistoryEvent> supplier) {
+    HistoryEvent historicInstance = supplier.get();
+
+    if (historicInstance != null) {
+      String rootProcessInstanceId = historicInstance.getRootProcessInstanceId();
+      authorization.setRootProcessInstanceId(rootProcessInstanceId);
+
+      Date removalTime = historicInstance.getRemovalTime();
+      authorization.setRemovalTime(removalTime);
+
+    } else { // reset
+      authorization.setRootProcessInstanceId(null);
+      authorization.setRemovalTime(null);
+
+    }
+  }
+
+  protected HistoryEvent getHistoricTaskInstance(CommandContext commandContext) {
+    String historicTaskInstanceId = authorization.getResourceId();
+
+    if (historicTaskInstanceId == null) {
```

Comment:
We can also skip the query if the resourceId is `*`

Reply:
👍